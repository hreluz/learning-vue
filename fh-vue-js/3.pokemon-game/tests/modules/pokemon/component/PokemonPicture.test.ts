import PokemonPicture from '@/modules/pokemon/components/PokemonPicture.vue';
import { mount } from '@vue/test-utils';

describe('<PokemonPicture />', () => {
  const imageSource = `https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/25.svg`;

  test('should match snapshot', () => {
    const wrapper = mount(PokemonPicture, {
      props: {
        showPokemon: false,
      },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  test('should show pokemon image', () => {
    const wrapper = mount(PokemonPicture, {
      props: {
        pokemonId: 25,
        showPokemon: true,
      },
    });

    const image = wrapper.find('img');

    expect(image.attributes('src')).toBe(imageSource);

    expect(image.attributes()).toEqual(
      expect.objectContaining({
        class: 'fade-in h-[200px]',
        src: imageSource,
        alt: '',
      }),
    );
  });

  test('should render the hidden image when showPokemon prop is false', () => {
    const wrapper = mount(PokemonPicture, {
      props: {
        showPokemon: false,
      },
    });

    expect(wrapper.find('img').classes('brightness-0')).toBe(true);
  });

  test('should render the hidden image when showPokemon prop is false', () => {
    const wrapper = mount(PokemonPicture, {
      props: {
        showPokemon: true,
      },
    });

    expect(wrapper.find('img').classes('fade-in')).toBe(true);
  });
});
