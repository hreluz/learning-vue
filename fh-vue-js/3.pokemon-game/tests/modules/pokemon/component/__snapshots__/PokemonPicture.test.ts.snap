// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<PokemonPicture /> > should match snapshot 1`] = `"<section data-v-49cc19b7=""><img data-v-49cc19b7="" src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/25.svg" class="brightness-0 h-[200px]" alt=""></section>"`;
